tests/tasks/test_notebook.py

```diff
@@ -167,124 +166,79 @@ def test_notebook_conversion_stores_as_unicode(tmp_directory, monkeypatch):
 def test_execute_sample_nb(name, out_dir, executor, tmp_sample_tasks):
     dag = DAG()
 
-    nb_source = NotebookSource(Path(name))
-
-    if executor not in ["ploomber-engine", "papermill"]:
-        with pytest.raises(ValueError) as excinfo:
-            NotebookRunner(
-                Path(name),
-                product=File(Path(out_dir, name + ".out.ipynb")),
-                dag=dag,
-                executor=executor,
-            )
-            dag.build()
-        assert "Invalid executor" in str(excinfo.value)
-    elif nb_source.language == "r" and executor == "ploomber-engine":
-        with pytest.raises(Exception) as excinfo:
-            NotebookRunner(
-                Path(name),
-                product=File(Path(out_dir, name + ".out.ipynb")),
-                dag=dag,
-                executor=executor,
-            )
-            dag.build()
-
-        assert "NameError: name 'c' is not defined" in str(excinfo.value)
-    else:
-        NotebookRunner(
-            Path(name),
-            product=File(Path(out_dir, name + ".out.ipynb")),
-            dag=dag,
-            executor=executor,
-        )
-        dag.build()
+    NotebookRunner(
+        Path(name),
+        product=File(Path(out_dir, name + ".out.ipynb")),
+        dag=dag,
+        executor=executor,
+    )
+    dag.build()
+
+
+class MyNullContext:
+    def __enter__(self, *args, **kwargs):
+        pass
+
+    def __exit__(self, *args, **kwargs):
+        pass
+
+
+does_not_raise = MyNullContext()
 
 
 @pytest.mark.parametrize(
-    "name, out_dir, executor",
+    "name, out_dir, executor, expectation, error_msg",
     [
-        ["sample.R", ".", "papermill"],
-        ["sample.R", ".", "ploomber-engine"],
-        ["sample.R", ".", "some-other-executor"],
+        ["sample.R", ".", "papermill", does_not_raise, None],
+        [
+            "sample.R",
+            ".",
+            "ploomber-engine",
+            pytest.raises(DAGBuildError),
+            "NameError: name 'c' is not defined",
```

Comment:
part of the AC was to improve the error message when trying to run an R notebook with ploomber-engine, has this been implemented? 

https://github.com/ploomber/ploomber/issues/1109#issuecomment-1600642210

Reply:
[Earlier Comment](https://github.com/ploomber/ploomber/pull/1123#issuecomment-1625590595)
> So, what is happening here, is there is no filter in ploomber-engine to detect the file extension type, so it always assume it is python and pass the code to InteractiveShell of Ipython. In this case the code passed is  c(1, 2, 3), which results in NameError: name 'c' is not defined. The ideal fix, I think would be to add file language detection to ploomber-engine

Reply:
yes, so part of the AC for this issue is precisely to improve the error message. Inside NotebookRunner, you can access the source property, which will return a NotebookSource (unsure if this is the exact name), I think there is a property there that return's the notebook's language. then, if you find it's an R notebook, you can raise an error if the user tries to use ploomber-engine